contracts/GoodGhostingPolygonCurve.sol

```diff
@@ -0,0 +1,674 @@
+            lpToken = IERC20(pool.lp_token());
+        } else if (_poolType == 1) {
+            lpToken = IERC20(pool.token());
+        }
+    }
+
+    /// @notice pauses the game. This function can be called only by the contract's admin.
+    function pause() external onlyOwner whenNotPaused {
+        _pause();
+    }
+
+    /// @notice unpauses the game. This function can be called only by the contract's admin.
+    function unpause() external onlyOwner whenPaused {
+        _unpause();
+    }
+
+    /// @notice Allows a player to join the game
+    function joinGame(uint256 _minAmount) external virtual whenNotPaused {
+        _joinGame(_minAmount);
+    }
+
+    /// @notice Allows a player to withdraws funds before the game ends. An early withdrawl fee is charged.
+    /// @dev Cannot be called after the game is completed.
+    function earlyWithdraw(uint256 _minAmount)
+        external
+        whenNotPaused
+        whenGameIsNotCompleted
+    {
+        Player storage player = players[msg.sender];
+        require(player.amountPaid > 0, "Player does not exist");
+        require(!player.withdrawn, "Player has already withdrawn");
+        player.withdrawn = true;
+        activePlayersCount = activePlayersCount.sub(1);
+        if (winnerCount > 0 && player.isWinner) {
+            winnerCount = winnerCount.sub(uint256(1));
+            player.isWinner = false;
+            winners[player.winnerIndex] = address(0);
+        }
+
+        // In an early withdraw, users get their principal minus the earlyWithdrawalFee % defined in the constructor.
+        uint256 withdrawAmount = player.amountPaid.sub(
+            player.amountPaid.mul(earlyWithdrawalFee).div(100)
+        );
+        // Decreases the totalGamePrincipal on earlyWithdraw
+        totalGamePrincipal = totalGamePrincipal.sub(player.amountPaid);
+        uint256 currentSegment = getCurrentSegment();
+
+        // Users that early withdraw during the first segment, are allowed to rejoin.
+        if (currentSegment == 0) {
+            player.canRejoin = true;
+        }
+
+        emit EarlyWithdrawal(msg.sender, withdrawAmount, totalGamePrincipal);
+
+        if (poolType == 0) {
+            uint256[numAaveTokens] memory amounts;
+            for (uint256 i = 0; i < numAaveTokens; i++) {
+                if (i == inboundTokenIndexUint) {
+                    amounts[i] = withdrawAmount;
+                } else {
+                    amounts[i] = 0;
+                }
+            }
+            uint256 poolWithdrawAmount = pool.calc_token_amount(amounts, true);
+
+            if (gauge.balanceOf(address(this)) < poolWithdrawAmount) {
+                poolWithdrawAmount = gauge.balanceOf(address(this));
+            }
+
+            gauge.withdraw(poolWithdrawAmount, false);
+
+            pool.remove_liquidity_one_coin(
+                poolWithdrawAmount,
+                inboundTokenIndexInt,
+                _minAmount,
+                true
+            );
```

Comment:
why we call `approve` when poolType is 1 (below - lines 308-311), but not here for poolType == 0?

Reply:
we discussed this on a tech call before the reason being the code of aave and atricrypto pool is completely different , in the case of aave i.e pool type 0 all funds sit in that contract, but atricrypto is in communication with other pools and funds sit in those pools hence the approval is needed because it is talking with external contracts